src/InferenceFramework.cpp

```diff
@@ -296,9 +296,11 @@ void ComputeKernelWindow(int nPar, const std::vector<particle>& particleList, do
 std::discrete_distribution<int> ComputeWeightDistribution(
 	const std::vector<EERAModel::particle>& particleList) {
 	
-	std::vector<double> weight_val;
-	for (auto p : particleList) {
-		weight_val.push_back(p.weight);
+	std::vector<double> weight_val(particleList.size(), 0.0);
+	int counter = 0;
+	for (const auto& p : particleList) {
```

Comment:
This seems like a messy compromise between two goals:
1. Use a range-based for loop
2. Preallocate the `weight_val` vector

I'd suggest simplifying it by switching back to an index-based loop:

```
	std::vector<double> weight_val(particleList.size());

	for (unsigned int i = 0; i < weight_val.size(); i++) {
		weight_val[i] = p.weight;
	}
```